flink-streaming-java/src/main/java/org/apache/flink/streaming/api/graph/StreamGraphGenerator.java

```diff
@@ -452,6 +457,33 @@ private boolean isUnboundedSource(final Transformation<?> transformation) {
             }
         }
 
+        transform
```

Comment:
Looks to me the `slotSharingGroupResources` can be different in cases that
1. a transformation without SlotSharingGroup set
2. above transformation is set with a "default" SlotSharingGroup

However, theoretically they are the same case.